lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py

```diff
@@ -0,0 +1,1312 @@
+            "barcode_swabsend",	# from husky_test_kit_registration
+            "return_utm_barcode", # from post_collection_data_entry_qc
+        }
+
+        for barcode_field in barcode_fields:
+            record[barcode_field] = record[barcode_field].strip().lower()
+
+        return record["collect_barcode_kiosk"] or record["return_utm_barcode"] or \
+            record["pre_scan_barcode"] or None
+
+
+    barcode = specimen_barcode()
+
+    if not barcode:
+        LOG.warning("Could not create Specimen Resource due to lack of barcode.")
+        return None, None
+
+    specimen_identifier = create_identifier(
+        system = f"{INTERNAL_SYSTEM}/sample",
+        value = barcode
+    )
+
+    # YYYY-MM-DD HH:MM:SS in REDCap
+    received_time = record['samp_process_date'].split()[0] if record.get('samp_process_date') else None
+
+    note = None
+
+    if record['able_to_test'] == 'no':
+        note = 'never-tested'
+    else:
+        note = 'can-test'
+
+    specimen_type = 'NSECR'  # Nasal swab.
+    specimen_resource = create_specimen_resource(
+        [specimen_identifier], patient_reference, specimen_type, received_time,
+        collection_date(record, collection_method), note
+    )
+
+    return create_entry_and_reference(specimen_resource, "Specimen")
+
+
+def collection_date(record: dict, collection_method: CollectionMethod) -> Optional[str]:
+    """
+    Determine sample/specimen collection date from the given REDCap *record*.
+    """
+    if collection_method == CollectionMethod.KIOSK:
+        return record["nasal_swab_q"]
+    elif collection_method == CollectionMethod.SWAB_AND_SEND:
+        return record["date_on_tube"] or record["kit_reg_date"] or record["back_end_scan_date"]
+    else:
+        return None
+
+def combine_multiple_fields(record: Dict[Any, Any], field_prefix: str, field_suffix: str = "") -> Optional[List]:
+        """
+        Handles the combining of multiple fields asking the same question such
+        as country and state traveled.
+        """
+        regex = rf'^{re.escape(field_prefix)}[0-9]+{re.escape(field_suffix)}$'
+        empty_value = ''
+        answered_fields = list(filter(lambda f: filter_fields(f, record[f], regex, empty_value), record))
+
+        if not answered_fields:
+            return None
+
+        return list(map(lambda x: record[x], answered_fields))
+
+def create_enrollment_questionnaire_response(record: dict, patient_reference: dict,
+                                            encounter_reference: dict) -> Optional[dict]:
+    """
+    Returns a FHIR Questionnaire Response resource entry for the enrollment
+    encounter (i.e. encounter of enrollment into the study)
+    """
+
+    # Do not include `core_age_years` because we calculate the age ourselves in the computed questionnaire.
+    integer_questions = [
+        'core_zipcode',
+        'weight',
+        'height_ft',
+        'height_in',
+        'height_ft_calc',
+        'height_total',
```

Comment:
These fields seem redundant, we can ingest one field (`height_total`?) and do conversions to ft/in when we need it.

Reply:
Fair point. I kept height_total.